### DISCUSSION TITLE
About getBlock

### DISCUSSION THREAD

**Discussion by puffanddmx** [crypto-org-chain/chain-main#568] [Q&A]
Hi
If i use block decoder for getting block with below request, it shows the human readable type.
But inside the block, there are no transaction hash information of each txs inside block.
How to get txs hash inside the block using block decoder?
Also if i do the block decoder for the message txs, there are info but format is not this type(from,to,amount).
How to extract that type using block decoder?

i want to get this info https://crypto.org/explorer/api/v1/blocks/832211/transactions?pagination=offset&page=1&limit=5 using sdk.

if i do curl curl "localhost:26657/tx_search?query=\"tx.height=832211\"&prove=true" -H "accept: application/json", i got the below error also.

```
request
await axios.get(${httpEndpoint}:${httpPort}+'/block?height=' +${height})

response
"data": {
"txs": [
{
"authInfo": {
"signerInfos": [
{
"sequence": {
"low": 0,
"high": 0,
"unsigned": true
},
"publicKey": {
"typeUrl": "/cosmos.crypto.secp256k1.PubKey",
"value": {
"0": 10,
"1": 33,
"2": 3,
"3": 76,
"4": 173,
"5": 13,
"6": 87,
"7": 174,
"8": 235,
"9": 215,
"10": 87,
"11": 2,
"12": 178,
"13": 187,
"14": 9,
"15": 55,
"16": 219,
"17": 105,
"18": 65,
"19": 199,
"20": 37,
"21": 171,
"22": 144,
"23": 111,
"24": 161,
"25": 202,
"26": 86,
"27": 96,
"28": 83,
"29": 196,
"30": 138,
"31": 38,
"32": 180,
"33": 72,
"34": 21
}
},
"modeInfo": {
"single": {
"mode": 127
}
}
}
],
"fee": {
"gasLimit": {
"low": 500000,
"high": 0,
"unsigned": true
},
"payer": "",
"granter": "",
"amount": [
{
"denom": "basecro",
"amount": "20000"
}
]
}
},
"body": {
"memo": "",
"timeoutHeight": {
"low": 0,
"high": 0,
"unsigned": true
},
"messages": [
{
"typeUrl": "/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward",
"value": "\n*cro1ey3dq34rv4n75t3znzv0rd3c0lx708fm70dc3a\u0012.crocncl1c04xdtwhy7f6h4ad4uy5cme6804m25r3c7yrk2"
},

<error>
curl "localhost:26657/tx_search?query=\"tx.height=832211\"&prove=true" -H "accept: application/json"
{
  "jsonrpc": "2.0",
  "id": -1,
  "error": {
    "code": -32603,
    "message": "Internal error",
    "data": "open chaindata/data/tx_index.db/234328.ldb: too many open files"
  }

```

**Comment by yihuang** [comment]
I think the endpoint [/tx_search?query="tx.height=_"](https://mainnet.crypto.org:26657/tx_search?query=%22tx.height=832211%22) is the way to go, can you open an issue for the `too many open files` error.

**Reply by tomtau** [reply]
For open files: https://docs.tendermint.com/master/tendermint-core/running-in-production.html
> Nodes should be given a sizable open file limit, e.g. 8192, via `ulimit -n 8192` or other deployment-specific mechanisms.
